src/server/game/Scripting/ScriptDefines/PlayerScript.cpp

```diff
@@ -895,6 +895,21 @@ bool ScriptMgr::AnticheatCheckMovementInfo(Player* player, MovementInfo const& m
     CALL_ENABLED_BOOLEAN_HOOKS(PlayerScript, PLAYERHOOK_ANTICHEAT_CHECK_MOVEMENT_INFO, !script->AnticheatCheckMovementInfo(player, movementInfo, mover, jump));
 }
 
+
+bool ScriptMgr::OnBeforePlayerResurrect(Player* player, float restorePercent, bool applySickness)
```

Comment:
All places that call the hook have to provide restorePercent and applySickness, regardless of the function actually using or providing these values.
If these values are not provided dummy values have to be used, which is less than ideal.